packages/inferno/src/DOM/utils/common.ts

```diff
@@ -80,12 +90,33 @@ export function findDOMfromVNode(vNode: VNode, startEdge: boolean) {
   return null;
 }
 
-export function removeVNodeDOM(vNode: VNode, parentDOM: Element) {
+export function callAllAnimationHooks(animationQueue: Function[], callback?: Function) {
+  let animsLeft = animationQueue.length;
+  do {
+    const fn = animationQueue.pop();
+    // This check shouldn't be needed but TS complains so adding it
+    if (fn !== undefined) {
+      fn(() => {
+        // When all animations are done, remove everything.
+        // INVESTIGATE: If we add a sibling when the animation is active, will it be removed?
+        if (--animsLeft <= 0 && isFunction(callback)) {
+          callback();
+        }
+      });
+    }
+  } while (animationQueue.length > 0);
+}
+
+function clearVNodeDOM(vNode: VNode, parentDOM: Element, deferedRemoval = false) {
```

Comment:
there should be no optional parameters in inferno js code because that goes in V8 engine trampoline routine which is slow

Reply:
Will fix